xarray/tests/test_backends.py

```diff
@@ -2213,6 +2212,77 @@ class PyNioTestAutocloseTrue(PyNioTest):
     autoclose = True
 
 
+@requires_netCDF4
+@requires_pnc
+class TestPseudoNetCDF(CFEncodedDataTest, NetCDF3Only, TestCase):
+    def test_write_store(self):
+        # pseudonetcdf is read-only for now
+        pass
+
+    @contextlib.contextmanager
+    def open(self, path, **kwargs):
+        with open_dataset(path, engine='pseudonetcdf',
+                          autoclose=self.autoclose,
+                          **kwargs) as ds:
+            yield ds
+
+    def save(self, dataset, path, **kwargs):
+        dataset.to_netcdf(path, engine='netcdf4', **kwargs)
+
+    def test_weakrefs(self):
+        example = Dataset({'foo': ('x', np.arange(5.0))})
+        expected = example.rename({'foo': 'bar', 'x': 'y'})
+
+        with create_tmp_file() as tmp_file:
+            example.to_netcdf(tmp_file, engine='netcdf4')
+            on_disk = open_dataset(tmp_file, engine='pseudonetcdf')
+            actual = on_disk.rename({'foo': 'bar', 'x': 'y'})
+            del on_disk  # trigger garbage collection
+            assert_identical(actual, expected)
+
+    @pytest.mark.skip(reason='cannot pickle file objects')
+    def test_pickle(self):
+        pass
+
+    @pytest.mark.skip(reason='cannot pickle file objects')
+    def test_pickle_dataarray(self):
```

Comment:
I'm not sure we should be skipping these tests.

They are skipped for `ScipyFileObjectTest` because it's not possible to pickle a `BytesIO()` object.

But it should work perfectly file to pickle the `opener()` function, which just keeps track of the file to open. If you use the model of the existing backend classes this should probably just work -- do you get an error message when you don't skip these?

Reply:
I do get an error message.

```
E       AttributeError: 'PncDataStore' object has no attribute '_mode'
```

Reply:
If I add `_mode`, then I start getting another error:

```
    def assert_open(self):
        if not self._isopen:
>           raise AssertionError('internal failure: file must be open '
                                 'if `autoclose=True` is used.')
E           AssertionError: internal failure: file must be open if `autoclose=True` is used.
```

Reply:
I figured out the pickle problem. So, this is no longer an issue.